opwen_email_server/actions.py

```diff
@@ -238,6 +239,38 @@ def _new_email_id(cls, email: str) -> str:
         return sha256(email.encode('utf-8')).hexdigest()
 
 
+class ProcessServiceEmail(_Action):
+    def __init__(self,
+                 raw_email_storage: AzureTextStorage,
+                 email_storage: AzureObjectStorage,
+                 next_task: Callable[[str], None],
+                 email_parser: Callable[[str], dict] = None,
+                 email_formatter: Callable[[str], dict] = None):
+
+        self._raw_email_storage = raw_email_storage
+        self._next_task = next_task
+        self._email_parser = email_parser or MimeEmailParser()
+        self._email_formatter = email_formatter or EmailFormatter()
```

Comment:
The design of this method is nice and re-usable: I believe we should be able to just pass in another email formatter to implement another email service. Good job!

To make this even more explicit, I'd recommend to make the `email_formatter` argument required instead of defaulting to this "echo" email formatter. In this way the code will make it clear that this action is a generic workflow for how to implement a "service via email handler" and the custom business logic for the service should be implemented in the formatter.

Reply:
Ok, sounds like a plan!